proto/cosmos/base/reflection/v1beta1/reflection.proto

```diff
@@ -21,6 +25,187 @@ service ReflectionService {
   };
 }
 
+// AppDescriptor describes a cosmos-sdk based application
+message AppDescriptor {
+  // chain provides the chain descriptor
+  ChainDescriptor chain = 2;
+  // codec provides metadata information regarding codec related types
+  CodecDescriptor codec = 3;
+  // configuration provides metadata information regarding the sdk.Config type
+  ConfigurationDescriptor configuration = 4;
+  // query_services provides metadata information regarding the available queriable endpoints
+  QueryServicesDescriptor query_services = 5;
+  // tx provides metadata information regarding how to send transactions to the given application
+  TxDescriptor tx = 6;
+}
+
+// TxDescriptor describes the accepted transaction type
+message TxDescriptor {
+  // fullname is the protobuf fullname of the raw transaction type (for instance the tx.Tx type)
+  string fullname = 1;
+  // authn provides information on how to authenticate a transaction in offline mode
+  AuthnDescriptor authn = 2;
+  // msgs lists the accepted application messages (sdk.ServiceMsg, sdk.Msg)
+  // NOTE: not to be confused with proto.Message types
+  repeated MsgDescriptor msgs = 3;
+}
+
+// AuthnDescriptor provides information on how to sign transactions without relying
+// on the online RPCs GetTxMetadata and CombineUnsignedTxAndSignatures
+message AuthnDescriptor {
+  // sign_modes defines the supported signature algorithm
+  repeated SigningModeDescriptor sign_modes = 1;
+}
+
+// SigningModeDescriptor provides information on a signing flow of the application
+// NOTE(fdymylja): here we could go as far as providing an entire flow on how
+// to sign a message given a SigningModeDescriptor, but it's better to think about
+// this another time
+message SigningModeDescriptor {
+  // name defines the unique name of the signing mode
+  string name = 1;
+  // authn_info_provider_method_fullname defines the fullname of the method to call to get
+  // the metadata required to authenticate using the provided sign_modes
+  string authn_info_provider_method_fullname = 2;
```

Comment:
What is this for?

Reply:
this is a left over of my attempts to not force reflection clients to rely on auth.

Example: I wanna get the metadata required to sign with sign_mode_direct, which are acc num, sequence and chainID then I can provide to the reflection client an endpoint to query for this info.

currently auth does not support this, although I feel it should. Or generally speaking auth should be abstracted better in the sdk.

Reply:
In general there should be a simple service to get acc num and sequence independent of how the auth module represents accounts. As you probably know, I think `x/auth` is way too complicated. But in the short term we should have a simple generic "get auth info" service somewhere...

Reply:
Yes, I wanted to implemented this specific endpoint, but I didn't know where to put it to be honest, should it be in auth? I don't think so. I think it should be in base and auth should implement it. But then it would be inconsistent with how we deal with authentication which is partly in base and partly in auth. I think this deserves an aside conversation which fits your discussion: https://github.com/cosmos/cosmos-sdk/discussions/9011.